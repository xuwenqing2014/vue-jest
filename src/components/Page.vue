<template>
  <div>
    <PageChild :messages="messages" />
  </div>
</template>

<script>
import PageChild from "./PageChild.vue";

export default {
  data() {
    return {
      messages: ["Cat"]
    };
  },
  components: {
    PageChild
  }
};
</script>

<style lang="scss" scoped>
</style>