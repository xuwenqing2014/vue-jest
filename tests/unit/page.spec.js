import {mount, shallowMount} from '@vue/test-utils';
import Page from '@/components/Page.vue';

describe('测试mount和shadow', () => {
    // shadowMount 不渲染子组件
    // mount会渲染子组件
    const wrapper = shallowMount(Page);
    // const wrapper = mount(Page);
    wrapper.setData({
        messages: ['aaa', 'bbb']
    });
    it('测试setData', () => {
        expect(wrapper.vm.messages).toEqual(['aaa', 'bbb'])
    });
    it('增加快照', () => {
        expect(wrapper.vm.$el).toMatchSnapshot();
    });
});