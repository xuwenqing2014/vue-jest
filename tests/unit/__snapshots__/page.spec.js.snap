// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试mount和shadow 增加快照 1`] = `
<div>
  <pagechild-stub
    messages="aaa,bbb"
  />
</div>
`;
